server/src/main/java/com/linecorp/centraldogma/server/EncryptionConfig.java

```diff
@@ -51,7 +58,37 @@ public EncryptionAtRestConfig(@JsonProperty("enabled") @Nullable Boolean enabled
     }
 
     /**
-     * Returns whether encryption at rest is enabled.
+     * Validates the encryption state with the data directory.
+     * This method checks if encryption was previously enabled and prevents disabling it.
+     *
+     * @param dataDir the data directory where the marker file is stored
+     * @throws IllegalStateException if encryption was previously enabled but is now disabled
+     */
+    void validateEncryptionState(File dataDir) {
+        requireNonNull(dataDir, "dataDir");
+        final Path markerFilePath = dataDir.toPath().resolve(ENCRYPTION_MARKER_FILE_NAME);
+        final boolean wasEncryptionEnabled = Files.exists(markerFilePath);
+
+        if (wasEncryptionEnabled && !enabled) {
+            throw new IllegalStateException(
+                    "Cannot disable encryption after it has been enabled. " +
+                    "Encryption was previously enabled. To proceed, you must keep encryption enabled.");
+        }
+
+        if (enabled && !wasEncryptionEnabled) {
+            // Create the marker file to indicate encryption has been enabled
+            try {
+                Files.createDirectories(dataDir.toPath());
+                Files.createFile(markerFilePath);
```

Comment:
Optional) What do you think of writing a brief description that this file does in the file itself so that an administrator does not accidentally delete it.

Reply:
Good suggestion. 👍